### PR TITLE
[SPARK-19590][pyspark][ML] Update the document for QuantileDiscretizer in pyspark

### PR DESCRIPTION
**Pull request by VinceShieh**
## What changes were proposed in this pull request?
This PR is to document the changes on QuantileDiscretizer in pyspark for PR:
https://github.com/apache/spark/pull/15428

## How was this patch tested?
No test needed

Signed-off-by: VinceShieh <vincent.xie@intel.com>

### CODE REVIEW

File: python/pyspark/ml/feature.py [hunk 1]
```diff
@@ -1178,7 +1178,17 @@ class QuantileDiscretizer(JavaEstimator, HasInputCol, HasOutputCol, JavaMLReadab
...
     `QuantileDiscretizer` takes a column with continuous features and outputs a column with binned
     categorical features. The number of bins can be set using the :py:attr:`numBuckets` parameter.
-    The bin ranges are chosen using an approximate algorithm (see the documentation for
+    It is possible that the number of buckets used will be less than this value, for example, if
+    there are too few distinct values of the input to create enough distinct quantiles.
+
+    NaN handling: Note also that
+    QuantileDiscretizer will raise an error when it finds NaN values in the dataset, but the user
+    can also choose to either keep or remove NaN values within the dataset by setting
+    `handleInvalid`. If the user chooses to keep NaN values, they will be handled specially and
```
Comment (holdenk) [inline python/pyspark/ml/feature.py]: could we maybe link this with a py attr like we did with numBuckets?

**Review by holdenk — COMMENTED** [review]
Thanks for following through and updating the Python documentation as well :) One minor question :)

File: python/pyspark/ml/feature.py (hunk 1 above)
Comment (VinceShieh) [inline python/pyspark/ml/feature.py]: yeah, sure. Thanks for pointing that out... ;)

**Comment by holdenk** [comment]
Thanks for doing the follow up Python work, merged to master :)

**Pull request closed without merging**